src/sentry/event_manager.py

```diff
@@ -690,6 +697,57 @@ def _pull_out_data(jobs: Sequence[Job], projects: ProjectsMapping) -> None:
         job["groups"] = []
 
 
+def _set_project_platform_if_needed(project: Project, event: Event) -> None:
+    # Only infer the platform if it's useful - if the event platform is "other" or null, there's
+    # no useful information for us to set the project platform
+    if not event.platform or event.platform == "other":
+        return
+
+    # Use a lock to prevent race conditions when multiple events are processed
+    # concurrently for a project with no initial platform
+    lock_key = f"project-platform-lock:{project.id}"
+
+    try:
+        with locks.get(lock_key, duration=60, name="project-platform-lock").acquire():
+            project.refresh_from_db(fields=["platform"])
+
+            if not project.platform:
+                with transaction.atomic(router.db_for_write(Project)):
+                    project.update(platform=event.platform)
```

Comment:
is it possible that the event is the example event? Should we filter those out? I believe we let people create an example event after creating a project